session/grpchijack/dial.go

```diff
@@ -111,6 +111,7 @@ func (c *conn) Close() (err error) {
 			err = c.stream.RecvMsg(m)
 			if err != nil {
 				if err != io.EOF {
+					c.readMu.Unlock()
```

Comment:
This was the cause of the failure.

CI now passing :tada:

@tonistiigi @thaJeztah PTAL

Reply:
I think you could equivalently have moved the existing `Unlock()`  to a `defer` next to the `Lock()`?